app.py

```diff
@@ -29,6 +35,65 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
             )
             aws_iam.PermissionsBoundary.of(self).apply(permission_boundary_policy)
 
+    def build_oidc(
+        self, oidc_provider_arn: str, oidc_repo_id: str, secret_arn: str, stage: str
+    ):
+        # Locate an IAM OIDC provider for the specified provider ARN
+        oidc_provider = aws_iam.OpenIdConnectProvider.from_open_id_connect_provider_arn(
+            self, "OIDCProvider", oidc_provider_arn
+        )
+        # create IAM role for provider access from specified repo
+        # the role will allow a github action in that repo
+        # to deploy resources and read a secret
+        oidc_role = aws_iam.Role(
+            self,
+            f"veda-backend-oidc-role-{stage}",
+            assumed_by=aws_iam.WebIdentityPrincipal(
+                oidc_provider.open_id_connect_provider_arn,
+                conditions={
+                    "StringEquals": {
+                        f"{oidc_provider.open_id_connect_provider_issuer}:sub": f"repo:{oidc_repo_id}"
+                    }
+                },
+            ),
+        )
+        oidc_role.add_to_policy(
+            aws_iam.PolicyStatement(
+                actions=["sts:AssumeRoleWithWebIdentity"],
+                resources=[oidc_provider_arn],
+            )
+        )
+
+        # Create an IAM policy statement that allows getting the secret value
+        get_secret_statement = aws_iam.PolicyStatement(
+            effect=aws_iam.Effect.ALLOW,
+            actions=["secretsmanager:GetSecretValue"],
+            resources=[secret_arn],
+        )
+
+        oidc_policy = aws_iam.Policy(
+            self,
+            f"veda-backend-oidc-policy-{stage}",
+            policy_name=f"veda-backend-oidc-policy-{stage}",
+            roles=[oidc_role],
+            statements=[get_secret_statement],
+        )
```

Comment:
the oidc role needs way more permission than just getting secrets, permissions for creating all the resources in the stack.

Reply:
The role was built for more permissive environments, but we should revisit this with SMCE and MCP environments in mind.